Production development

urshala opened the pull request:
* Allow the project to be run on production environment. 
This requires `env.production` file for the KSQL credentials 
along with `.secrets` folder for all the certificates.

suutari-ai committed:
ta120_adapter: Refactor serializers to separate module

suutari-ai committed:
Specify datastream id in "observations" topic

Specify the datastream id instead of the property name in the
observations messages.

suutari-ai committed:
Add test scripts to create and emulate ta120Sensor

Add script to create ta120Sensor and send the readings to Kafka.

urshala committed:
Add configurations for Production environment

Allow the script to run the project on different environment.
And therefore add the production configurations.

urshala committed:
Add *.production file to .gitignore

suutari-ai commented on docker-compose.yml:
```diff
@@ -70,16 +70,18 @@ services:
       KSQL_CONFIG_DIR: "/etc/ksql"
       KSQL_LOG4J_OPTS: >-
         -Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties
-      KSQL_BOOTSTRAP_SERVERS: "kafka:9092"
       KSQL_HOST_NAME: ksql-server
       KSQL_APPLICATION_ID: "cp-all-in-one"
       KSQL_LISTENERS: "http://ksql-server:8088"
       KSQL_CACHE_MAX_BYTES_BUFFERING: 0
-      KSQL_KSQL_SCHEMA_REGISTRY_URL: "http://schema-registry:8081"
```
Can these still be here and be overridden in the `env.production` file?

suutari-ai commented on docker-compose.yml:
```diff
@@ -70,16 +70,18 @@ services:
         io.confluent.monitoring.clients.interceptor.MonitoringProducerInterceptor
       KSQL_CONSUMER_INTERCEPTOR_CLASSES: >-
         io.confluent.monitoring.clients.interceptor.MonitoringConsumerInterceptor
+    volumes:
+      - ./.secrets:/.secrets
```
Can/should we use the "secrets" feature of docker-compose for this? See https://docs.docker.com/compose/compose-file/#secrets

suutari-ai commented on env.development:
```diff
@@ -1 +1,4 @@
 DEBUG=True
+
+KSQL_BOOTSTRAP_SERVERS=kafka:9092
+KSQL_KSQL_SCHEMA_REGISTRY_URL=http://schema-registry:8081
```
As said above, I would like these to be in the compose file if possible.

suutari-ai commented on run_project.sh:
```diff
@@ -0,0 +1,11 @@
+else
+    echo "Running project in development environment."
+    docker-compose up
+fi
```
This feels a bit hacky solution... I hope we come up with something better...

suutari-ai commented:
* It would be useful to have a template for the `env.production` file (`env.production.template`) with the required configuration options and some dummy values for them. Can you add that?
* Also general instructions on how to do deploy it to the production are needed.

urshala committed:
Change the datahubhel.core models

The models in the datahubhel.core needs some refactoring to
comply with the STA. Therefore add the Location model and
attach it to Thing. Also, the Sensor model no longer serves the
purpose of physical sensor. It is more like a parent umbrella
under which there are other sensors. E.g there could be several
Ta120sensor of type Sensor.

urshala committed:
Add datastream to schema

Add datastream to schema for Avro.

urshala committed:
Change the ta120_adapter models

The Thing, ThingLocation and Location are no longer needed in
ta120_adapter. These should be maintained in core app. Therefore,
remove these from here. Model Ta120Sensor which  is the
representation of physical sensor.

urshala committed:
Make sure that many2many field on TA120Sensor get saved

The datastream needs to be attached to each TA120Sensor upon
creation. Since datastream is the M2M field, it threfore needs
to exist before it can be attached to TA120Sensor. This is why
we needed the transaction.commit so that the creating and attaching
of datastreams to TA120Sensor works on post_save.

urshala committed:
Create a model for Observation

Create a model for Observation. The data for this models comes
from Kafka. Therefore, it is generated using inspectdb command.

urshala committed:
Add KSQL scripts related to ta120Adapter

Add the necessary scripts that loads the connectors and creates
streams for ta120Adapter.

urshala committed:
WIP docker-secrets
